okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -237,6 +237,12 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
     val stream: Http2Stream
     val streamId: Int
 
+    // trigger interrupts before we write to the buffered output
+    // since flush is likely where we get notified.
+    if (Thread.interrupted()) {
+      throw InterruptedIOException()
+    }
+
```

Comment:
@swankjesse I do wonder whether we should have a try catch around everything here and send a RST_STREAM if we think we may have created a stream.  

Or tear down the entire connection at this point to recover?

Reply:
Made this change to tear down the connection.